src/main/java/com/group6/harmoniq/models/User.java

```diff
@@ -0,0 +1,129 @@
+package com.group6.harmoniq.models;
+
+import java.util.Date;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.persistence.Temporal;
+import jakarta.persistence.TemporalType;
+
+@Entity
+@Table(name = "users")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long uid;
+
+    // Enforce SpotifyID uniqueness and ensures every user has Spotify user ID
+    @Column(unique = true, nullable = false)
+    private String spotifyId;
+    
+    private String displayName;
+
+    private String email;
+
+    private int followers;
+
+    private int quizScore;
+
+    private String imageUrl;
+
+
+    // Default is regular user
+    private Boolean isAdmin = false;
+    
+    @Column(updatable = false)
+    @Temporal(TemporalType.TIMESTAMP)
+    private final Date createdAt = new Date();
+
+    // Constructors
+    public User() {}
+
+    public User(String spotifyId, String displayName, String email, int followers, int quizScore,  String imageUrl, Boolean isAdmin) {
+        this.spotifyId = spotifyId;
+        this.displayName = displayName;
+        this.email = email;
+        this.followers = followers;
+        this.quizScore = quizScore;
+        this.imageUrl = imageUrl;
+
+        this.isAdmin = isAdmin != null ? isAdmin : false; // Ensures admin is not null but still defaults to false
+    }
+
+    public Long getUId() {
```

Comment:
Because we're using the primitive type `long` for the property, not `Long` object
```suggestion
    public long getUId() {
```